projects/cdk/directives/auto-focus/autofocus.directive.ts

```diff
@@ -40,49 +43,70 @@ export class TuiAutoFocusDirective implements AfterViewInit {
         private readonly tuiFocusableComponent: TuiFocusableElementAccessor | null,
         @Inject(TUI_IS_IOS) private readonly isIos: boolean,
         @Inject(ANIMATION_FRAME) private readonly animationFrame$: Observable<number>,
+        @Inject(NgZone)
+        private readonly ngZone: NgZone,
     ) {}
 
     ngAfterViewInit() {
         if (!this.autoFocus) {
             return;
         }
 
-        const element =
-            this.tuiFocusableComponent?.nativeFocusableElement ||
-            this.elementRef.nativeElement;
-
         if (this.isIos) {
```

Comment:
I think we only need the hack for `input` and `textarea`, how about we add the check in the `if`?

Reply:
fixed